src/components/Switzerland/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { DistributionType, getVariantDistributionData } from '../../services/api';
+import { TimeZipCodeDistributionEntry } from '../../services/api-types';
+import { AccountService } from '../../services/AccountService';
+import * as zod from 'zod';
+import { SampleSelectorSchema } from '../../helpers/sample-selector';
+import { useResizeDetector } from 'react-resize-detector';
+
+import Map from './Map';
+
+const PropsSchema = SampleSelectorSchema;
+type Props = zod.infer<typeof PropsSchema> & { width?: number };
+
+const Switzerland = ({ country, mutations, matchPercentage }: Props) => {
+  const [distributionData, setDistributionData] = useState<TimeZipCodeDistributionEntry[]>([]);
+  const loggedIn = AccountService.isLoggedIn();
```

Comment:
I think this login check makes more sense in `FocusPage`.

Reply:
Ah, I just saw that this component also shows "Please log in" - now this makes more sense. However I think it's probably better just to hide the map compeletely (and not show a message) if the user is logged out, since very few users will ever be able to log in. What do you think @chaoran-chen?

Reply:
Later we should take it out but right now, I like the message so that other users know that our tool can do even more amazing things.